setup.py

```diff
@@ -25,22 +27,63 @@
 package_name = 'cross_compile'
 
 
+def build_package_directory_path():
+    """
+    Return the directory containing the package code, following symbolic links.
+
+    When compiling this package with colcon, users can optionally pass
+    a --symlink-install flag which will create a symlink of setup.py into
+    <prefix>/build/<pkg>.
+    Using this function, a path will be returned containing not only setup.py,
+    and some subset of files for which a symlink exists, but the original
+    package directory.
+    """
+    try:
+        return os.path.dirname(os.readlink(__file__))
+    except OSError as os_error:
+        if os_error.errno == errno.EINVAL:
+            return os.path.dirname(__file__)
+
+
 def read_version_from_package_xml():
     tree = ElementTree.parse('package.xml')
     version = tree.find('version')
     return version.text
 
 
+this_directory = build_package_directory_path()
+with open(path.join(this_directory, 'README.md'), encoding='utf-8') as f:
+    long_description = f.read()
```

Comment:
Unlikely but, it would be good to have a default so if it fails, `long_description` is still defined somewhere instead of throwing an error.

Reply:
Hiding failures is bad, if there is no README, there is a bug, and if there is a bug, we'll go and fix it.